flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/ExecutionContext.java

```diff
@@ -391,55 +431,119 @@ private static Executor lookupExecutor(
 		}
 	}
 
-	private void initializeTableEnvironment() {
+	private void initializeTableEnvironment(SessionState sessionState) {
 		//--------------------------------------------------------------------------------------------------------------
 		// Step.1 Create environments
 		//--------------------------------------------------------------------------------------------------------------
+		createTableEnvironment(sessionState);
+
+		final boolean noInheritedState = sessionState == null;
+		if (noInheritedState) {
```

Comment:
these initializations could happen when you create the `catalogManager`, `moduleManager` and `functionCatalog`, around line 478-488

Reply:
not sure about this anymore...seems you need the table env to initialize tables and other stuff..

Reply:
Yes, but we indeed can merge these 2 methods.